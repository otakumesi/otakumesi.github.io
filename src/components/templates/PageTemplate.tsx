import * as React from 'react'
import styled from 'styled-components'

const PageWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const ContentWrapper = styled.div`
  padding: 10px 0;
  margin: 0 auto;
`

export interface Props {
  header: JSX.Element
  footer: JSX.Element
  children: JSX.Element[] | JSX.Element
}

const PageTemplate = ({header, footer, children, ...props} : Props) => {
  return (
    <PageWrapper {...props}>
      {header}
      <ContentWrapper>
        {children}
      </ContentWrapper>
      {footer}
    </PageWrapper>
  )
}

export default PageTemplate
