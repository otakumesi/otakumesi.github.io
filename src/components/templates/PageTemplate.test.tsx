import * as React from 'react'
import PageTemplate from './PageTemplate'
import * as renderer from 'react-test-renderer'

const HeaderMock = () => (<React.Fragment>Header</React.Fragment>)
const FooterMock = () => (<React.Fragment>Footer</React.Fragment>)
const ContentMock = () => (<React.Fragment>Content</React.Fragment>)

describe('PageTemplate', () => {
  test('shows Header, Footer, and children', () => {
    const component = renderer.create(
      <PageTemplate header={<HeaderMock />} footer={<FooterMock />}>
        <ContentMock />
      </PageTemplate>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
