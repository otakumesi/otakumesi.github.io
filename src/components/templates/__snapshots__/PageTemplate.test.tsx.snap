// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplate shows Header, Footer, and children 1`] = `
<div
  className="sc-bdVaJa bPGLpq"
>
  Header
  <div
    className="sc-bwzfXH gPgjRf"
  >
    Content
  </div>
  Footer
</div>
`;
